code/optimize/src/test/java/com/adobe/marketing/mobile/optimize/OptimizeTests.java

```diff
@@ -90,475 +69,548 @@ public void test_extensionVersion() {
 
     @Test
     public void test_registerExtension() {
-        // test
-        Optimize.registerExtension();
-        final ArgumentCaptor<ExtensionErrorCallback<ExtensionError>> callbackCaptor = ArgumentCaptor.forClass(ExtensionErrorCallback.class);
-
-        // The Optimize extension should successfully register with Mobile Core
-        PowerMockito.verifyStatic(MobileCore.class, Mockito.times(1));
-        MobileCore.registerExtension(eq(OptimizeExtension.class), callbackCaptor.capture());
-
-        // verify the callback
-        ExtensionErrorCallback<ExtensionError> extensionErrorCallback = callbackCaptor.getValue();
-        assertNotNull("The extension error callback should not be null.", extensionErrorCallback);
+        try (MockedStatic<MobileCore> mobileCoreMockedStatic = Mockito.mockStatic(MobileCore.class)) {
+            // setup
+            ArgumentCaptor<Class> extensionClassCaptor = ArgumentCaptor.forClass(Class.class);
+            ArgumentCaptor<ExtensionErrorCallback> callbackCaptor = ArgumentCaptor.forClass(
+                    ExtensionErrorCallback.class
+            );
+            mobileCoreMockedStatic
+                    .when(() -> MobileCore.registerExtension(extensionClassCaptor.capture(), callbackCaptor.capture()))
+                    .thenReturn(true);
+            // test
+            Optimize.registerExtension();
+
+            // verify: happy
+            assertNotNull(callbackCaptor.getValue());
+            assertEquals(OptimizeExtension.class, extensionClassCaptor.getValue());
+            // verify: error callback was called
+            callbackCaptor.getValue().error(ExtensionError.UNEXPECTED_ERROR);
```

Comment:
nit: This `test_registerExtension()` should ideally be happy path test.  So test for `callbackCaptor.getValue().error(null);` in this test.